src/main/java/frc/robot/subsystems/ball/feeder/UltrasonicPoller.java

```diff
@@ -10,21 +10,63 @@
 
     Ultrasonic ultrasonicSensor;
 
-    UltrasonicPoller(Ultrasonic sensor, int time) {
+    // Consider renaming, couldn't think of a better one ATM
+    boolean noBalls;
+    long time;
+    double distance;
+    double currentRange;
+    int count;
+
+    long startTime;
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance, long rate) {
         // sensor is pretty self explanatory
         // time is how long you want the sensor to not detect an object until true is returned
-        super(25, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
+        // distance is the minimum distance threshold
+        super(rate, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
         this.ultrasonicSensor = sensor;
+        this.time = time;
+        this.distance = distance;
+        this.noBalls = false;
+        this.startTime = 0;
+        this.count = 0; // There has to be a better way to do this
+
+    }
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance) {
+        this(sensor, time, distance, 25);
     }
 
     @Override
     public void defineResources(){
         require(ultrasonicSensor);
     }
 
+    public void setTime(long time) {
+        this.time = time;
+    }
+
+    public void setDistance(double distance){
+        this.distance = distance;
+    }
+
+    // Consider renaming, probably want to generalize the name if we ever re-use it
+    public boolean checkUnloaded(){
+        return this.noBalls;
+    }
+
     @Override
     public void task(){
-        // Placholder
-        return;
+        this.currentRange = this.ultrasonicSensor.getRangeMM();
+        if(this.count == 0){ // TODO look into a better way of doing this
+            this.startTime = System.nanoTime(); // I know about a do while, but this isn't contained within a while loop 
+            this.count = 1;
+        }
+        if((this.currentRange > this.distance) && (((System.nanoTime() - this.startTime) / 100000) > this.time)) {
```

Comment:
Rather than dividing by `100000` (did you mean to use one hundred thousand here?), for better readability or ease of changing it in the future, please use something like `TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - this.startTime()) > time`

Reply:
I meant one-million. Irrelevant now anyways since I'll just use toMillis.